pyrasterframes/src/main/python/pyrasterframes/rf_types.py

```diff
@@ -349,9 +349,102 @@ def __div__(self, right):
         return self.__truediv__(right)
 
     def dimensions(self):
-        # list of cols, rows as is conventional in GeoTrellis and RasterFrames
+        """ Return a list of cols, rows as is conventional in GeoTrellis and RasterFrames."""
         return [self.cells.shape[1], self.cells.shape[0]]
 
+    def to_png(self):
+        """ Provide image of Tile."""
+        if self.cells is None:
+            return None
+
+        import io
+        from matplotlib.backends.backend_agg import FigureCanvasAgg as FigureCanvas
+        from matplotlib.figure import Figure
+
+        # Set up matplotlib objects
+        [width, height] = self.dimensions()
+        dpi = 300
+        nominal_size = 5.5  # approx full size for a 256x256 tile
+        fig = Figure(figsize=(nominal_size * width / dpi,   nominal_size * height / dpi))
+        canvas = FigureCanvas(fig)
+        axis = fig.add_subplot(1, 1, 1)
+
+        axis.imshow(self.cells)
+        axis.set_aspect('equal')
+        axis.xaxis.set_ticks([])
+        axis.yaxis.set_ticks([])
+
+        axis.set_title('{}, {})'.format(self.dimensions(), self.cell_type.__repr__()))  # compact metadata as title
+
+        with io.BytesIO() as output:
+            canvas.print_png(output)
+            return output.getvalue()
```

Comment:
If we're truly rendering to PNG, it should happen in the JVM where it's a lot faster.

Reply:
Seems like there's a lot more going on than just rendering a PNG... what's with the `Figure` and `axis` settings?

Reply:
Ahh i didn't realize we could do it on JVM side, but now i remember we used to do something like that... 

The Figure and axis are more about trying to get a certain output size and a particular styling with matplotlib, basically removing axis ticks and labels.

Reply:
The more I think about it the more I realize that it may not make sense to render PNG on the JVM... Because we only want to render PNG for the python `Tile` after we have serialized it from Catalyst.  Does that make sense?

Reply:
Yes.